tests/platform_helper/test_command_codebase.py

```diff
@@ -30,169 +26,49 @@ def mock_aws_client(get_aws_session_or_abort):
-        subprocess.run(
-            ["git", "remote", "add", "origin", "git@github.com:uktrade/some-test-app.git"],
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-        )
-
-        result = CliRunner().invoke(prepare)
-
-        assert (
-            "You are in the deploy repository; make sure you are in the application codebase repository."
-            in result.stdout
-        )
-        assert result.exit_code == 1
-
-    def test_codebase_prepare_generates_an_executable_image_build_run_file(self, tmp_path):
-        from dbt_platform_helper.commands.codebase import prepare
-
-        os.chdir(tmp_path)
-
-        subprocess.run(["git", "init"], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-        subprocess.run(
-            ["git", "remote", "add", "origin", "git@github.com:uktrade/another-test-app.git"],
-            stdout=subprocess.PIPE,
-            stderr=subprocess.PIPE,
-        )
-
-        result = CliRunner().invoke(prepare)
-
-        assert result.exit_code == 0
-        assert stat.filemode(Path(".copilot/image_build_run.sh").stat().st_mode) == "-rwxr--r--"
 
 
 class TestCodebaseBuild:
-    @patch("click.confirm")
-    @patch("subprocess.run")
-    @patch("dbt_platform_helper.commands.codebase.get_aws_session_or_abort")
-    def test_codebase_build_successfully_triggers_a_pipeline_based_build(
-        self, get_aws_session_or_abort, mock_subprocess_run, mock_click_confirm
+    @patch("dbt_platform_helper.commands.codebase.Codebase")
+    def test_codebase_build_does_not_trigger_build_without_an_application(
+        self, mock_codebase_object
     ):
-        from dbt_platform_helper.commands.codebase import build
-
-        mock_subprocess_run.return_value.stderr = ""
-        mock_click_confirm.return_value = "y"
 
-        client = mock_aws_client(get_aws_session_or_abort)
-        client.start_build.return_value = {
-            "build": {
-                "arn": "arn:aws:codebuild:eu-west-2:111111111111:build/build-project:build-id",
-            }
-        }
+        mock_codebase_object_instance = mock_codebase_object.return_value
+        mock_codebase_object_instance.build.side_effect = click.Abort
+        os.environ["AWS_PROFILE"] = "foo"
 
         result = CliRunner().invoke(
             build,
             [
                 "--app",
-                "test-application",
+                "not-an-application",
                 "--codebase",
                 "application",
                 "--commit",
                 "ab1c23d",
             ],
         )
 
-        client.start_build.assert_called_with(
-            projectName="codebuild-test-application-application",
-            artifactsOverride={"type": "NO_ARTIFACTS"},
-            sourceVersion="ab1c23d",
+        # Assert
+        mock_codebase_object_instance.build.assert_called_once_with(
+            "not-an-application", "application", "ab1c23d"
         )
+        assert result.exit_code == 1
```

Comment:
Isn't this behaviour the responsibility of `domain.Codebase` now?

So all we would want to be testing here is that `domain.Codebase.build` was called with the correct arguments?